tf-psa-crypto/drivers/builtin/src/sha256.c

```diff
@@ -617,7 +617,7 @@ static size_t mbedtls_internal_sha256_process_many(mbedtls_sha256_context *ctx,
     }
 }
 
-static int mbedtls_internal_sha256_process(mbedtls_sha256_context *ctx,
+int mbedtls_internal_sha256_process(mbedtls_sha256_context *ctx,
```

Comment:
If you remove the code lines 417 to 423 and lines 477 to 483 you should be able to make it static I'd say. Did you try that?

Reply:
Note that in https://github.com/Mbed-TLS/mbedtls/pull/9189, to be merged very soon, we turn on `-Wmissing-prototypes` in many builds, so functions without a prototype must be static.

Reply:
> If you remove the code lines 417 to 423 and lines 477 to 483 you should be able to make it static I'd say. Did you try that?

@ronald-cron-arm Unless I have misunderstood here, then I am afraid this doesn't seem to work.

Reply:
@tom-daubney-arm Please see https://github.com/Mbed-TLS/mbedtls/pull/9402/files/79ca53578627e8d7419cf334fc92fb6dce757c72..e60dca2352dcea2372873457fcae48002caee216#r1699814608.